x-pack/plugins/ui_actions_enhanced/public/drilldowns/components/connected_flyout_manage_drilldowns/connected_flyout_manage_drilldowns.tsx

```diff
@@ -250,108 +242,3 @@ export function createFlyoutManageDrilldowns({
     }
   };
 }
-
-function useCompatibleActionFactoriesForCurrentContext<
```

Comment:
didn't see a reason moving it to separate file :D. enjoyed how it's all centralised. :(
aren't asking to get it back.. just saying.

Reply:
The `connected_flyout_manage_drilldowns.tsx` file is growing too large and I found confusing to navigate in it, especially when React hooks are in the same file and are defined after the definition of component which uses them.

I think we should further clean up this file, ideally it would be just two lines like this:

```tsx
    if (route ===  Routes.Manage) return <ConnectedFlyoutListManageDrilldowns />;
    return <ConnectedFlyoutDrilldownWizard />;
```

instead of this *SwitchStatement*:

```tsx

    switch (route) {
      case Routes.Create:
      case Routes.Edit:
        return (
          <FlyoutDrilldownWizard
            docsLink={docsLink}
            triggerPickerDocsLink={triggerPickerDocsLink}
            showWelcomeMessage={shouldShowWelcomeMessage}
            onWelcomeHideClick={onHideWelcomeMessage}
            drilldownActionFactories={actionFactories}
            onClose={props.onClose}
            mode={route === Routes.Create ? 'create' : 'edit'}
            onBack={isCreateOnly ? undefined : () => setRoute(Routes.Manage)}
            onSubmit={({ actionConfig, actionFactory, name, selectedTriggers }) => {
              if (route === Routes.Create) {
                createDrilldown(
                  {
                    name,
                    config: actionConfig,
                    factoryId: actionFactory.id,
                  },
                  selectedTriggers
                );
              } else {
                editDrilldown(
                  currentEditId!,
                  {
                    name,
                    config: actionConfig,
                    factoryId: actionFactory.id,
                  },
                  selectedTriggers
                );
              }

              if (isCreateOnly) {
                if (props.onClose) {
                  props.onClose();
                }
              } else {
                setRoute(Routes.Manage);
              }

              setCurrentEditId(null);
            }}
            onDelete={() => {
              deleteDrilldown(currentEditId!);
              setRoute(Routes.Manage);
              setCurrentEditId(null);
            }}
            actionFactoryPlaceContext={props.placeContext}
            initialDrilldownWizardConfig={resolveInitialDrilldownWizardConfig()}
            supportedTriggers={props.triggers}
            getTrigger={getTrigger}
          />
        );

      case Routes.Manage:
      default:
        // show trigger column in case if there is more then 1 possible trigger in current context
        const showTriggerColumn =
          intersection(
            props.triggers,
            actionFactories
              .map((factory) => factory.supportedTriggers())
              .reduce((res, next) => res.concat(next), [])
          ).length > 1;
        return (
          <FlyoutListManageDrilldowns
            docsLink={docsLink}
            showWelcomeMessage={shouldShowWelcomeMessage}
            onWelcomeHideClick={onHideWelcomeMessage}
            drilldowns={drilldowns.map(mapToDrilldownToDrilldownListItem)}
            onDelete={(ids) => {
              setCurrentEditId(null);
              deleteDrilldown(ids);
            }}
            onEdit={(id) => {
              setCurrentEditId(id);
              setRoute(Routes.Edit);
            }}
            onCreate={() => {
              setCurrentEditId(null);
              setRoute(Routes.Create);
            }}
            onClose={props.onClose}
            showTriggerColumn={showTriggerColumn}
          />
        );
    }
```